pkg/storage/config/schema_config.go

```diff
@@ -305,17 +305,15 @@ func usingForPeriodConfigs(configs []PeriodConfig, fn func(PeriodConfig) bool) b
 	return false
 }
 
-func UsingObjectStorageIndex(configs []PeriodConfig) bool {
-	fn := func(cfg PeriodConfig) bool {
-		switch cfg.IndexType {
-		case BoltDBShipperType, TSDBType:
-			return true
-		default:
-			return false
-		}
-	}
+// UsingObjectStoreIndex returns true if the period config uses either boltdb-shipper or tsdb index.
+func UsingObjectStoreIndex(cfg PeriodConfig) bool {
+	return cfg.IndexType == BoltDBShipperType || cfg.IndexType == TSDBType
+}
 
-	return usingForPeriodConfigs(configs, fn)
+// ContainsObjectStorageIndex returns true if the current or any of the upcoming periods
+// use an object store index.
+func ContainsObjectStorageIndex(configs []PeriodConfig) bool {
```

Comment:
I called it `UsingObjectStoreIndex` because it checked whether current or upcoming index is an object storage based index. This new name makes it confusing. Maybe swap `UsingObjectStoreIndex` and `ContainsObjectStorageIndex` names?